airflow/utils/context.pyi

```diff
@@ -37,11 +37,13 @@ from airflow.models.dagrun import DagRun
 from airflow.models.dataset import DatasetEvent
 from airflow.models.param import ParamsDict
 from airflow.models.taskinstance import TaskInstance
-from airflow.serialization.pydantic.dag_run import DagRunPydantic
-from airflow.serialization.pydantic.dataset import DatasetEventPydantic
-from airflow.serialization.pydantic.taskinstance import TaskInstancePydantic
 from airflow.typing_compat import TypedDict
 
+if TYPE_CHECKING:
+    from airflow.serialization.pydantic.dag_run import DagRunPydantic
+    from airflow.serialization.pydantic.dataset import DatasetEventPydantic
+    from airflow.serialization.pydantic.taskinstance import TaskInstancePydantic
```

Comment:
Not needed since `pyi` is only loaded in the typing context in the first place (i.e. `TYPE_CHECKING` is implied)

Reply:
Nice!